File: fc-large2/ticketing/c-spring-cache/src/main/java/com/example/cspringcache/controller/UserController.java
package com.example.cspringcache.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

import com.example.cspringcache.domain.entity.RedisHashUser;
import com.example.cspringcache.domain.entity.User;
import com.example.cspringcache.domain.service.UserService;

import lombok.RequiredArgsConstructor;

@RestController
@RequiredArgsConstructor
public class UserController {
    private final UserService userService;

    @GetMapping("/users/{id}")
    public User getUser(@PathVariable Long id) {
        return userService.getUser3(id);
    }

    @GetMapping("/redishash-users/{id}")
    public RedisHashUser getUser2(@PathVariable Long id) {
        return userService.getUser2(id);
    }
}

File: 멀티모듈/제미니/api/src/main/kotlin/com/ys/api/controller/UserController.kt
package com.ys.api.controller

import com.ys.domain.user.UserService
import org.springframework.web.bind.annotation.*

@RestController
class UserController(
    private val userService: UserService
) {

    @PostMapping("Lusers")
    fun addUser(
        @RequestBody request: NewUserRequest
    ): NewUserResponse {

        return NewUserResponse(userService.add(request.name))
    }

    @GetMapping("/users/fuserIdl")
    fun findUser(
        @PathVariable userId: Long
    ): UserResponse {
        return UserResponse(userService.read(userId))

    }

}